### PR TITLE
Implementing function to chain hash table, store hash in a csv, update hash on shadowrite, and hash consistency checker

### PR DESCRIPTION
**Pull request by mnhn329**
Addresses last task of #11.

### CODE REVIEW

File: src/main/java/org/springframework/samples/petclinic/owner/OwnerRepositoryCSV.java [hunk 1]
```diff
@@ -97,6 +95,25 @@ public Owner save(String firstName, String lastName, String address, String city
...
 	}
 	
+    
+    public void storeHashRecord() {//stores/update hashrecord
+    	String filename ="hash-record/owners.csv";
+        try {
+            FileWriter fw = new FileWriter(filename);
+            Class.forName("com.mysql.jdbc.Driver").newInstance();
+            HashGenerator hash = new HashGenerator();
+    		String hashContent = hash.computeHash(OwnerController.appendHashedRows());//second hash
```
Comment (amawai) [inline src/main/java/org/springframework/samples/petclinic/owner/OwnerRepositoryCSV.java]: Is there a reason that`appendedHashedRows()` is in OwnerController? I think it should be contained within its own Repository class (i.e. OwnerRepositoryCSV) for cohesion purposes

File: src/main/java/org/springframework/samples/petclinic/owner/OwnerRepositoryCSV.java [hunk 2]
```diff
@@ -97,6 +95,25 @@ public Owner save(String firstName, String lastName, String address, String city
 		return owner;
 	}
 	
+    
+    public void storeHashRecord() {//stores/update hashrecord
+    	String filename ="hash-record/owners.csv";
+        try {
+            FileWriter fw = new FileWriter(filename);
+            Class.forName("com.mysql.jdbc.Driver").newInstance();
```
Comment (amawai) [inline src/main/java/org/springframework/samples/petclinic/owner/OwnerRepositoryCSV.java]: What is the purpose of this?

File: src/main/java/org/springframework/samples/petclinic/owner/PetController.java [hunk 3]
```diff
@@ -186,6 +191,27 @@ public void forklift() {
         }
     }
 
+    public static String appendHashedRows() {//collects rows, hashes them, appends them and returns
```
Comment (amawai) [inline src/main/java/org/springframework/samples/petclinic/owner/PetController.java]: Same comment about the equivalent function in OwnerController

File: src/main/java/org/springframework/samples/petclinic/owner/ConsistencyChecker.java [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+						"\n\t changes found in owners CSV");
+		    	inconsistencies++;
+		    }
+			
+			hashReader.close();
+			
+			if (inconsistencies == 0) 
+            	System.out.println("No inconsistencies across former owners table dataset.");
+			
+			log.info("The time is now {}", dateFormat.format(new Date()));
```
Comment (amawai) [inline src/main/java/org/springframework/samples/petclinic/owner/ConsistencyChecker.java]: Is this still necessary?

**Review by amawai — CHANGES_REQUESTED** [review]
Looks good, however, please address comments regarding cohesion and remove unnecessary MySQL connections

File: src/main/java/org/springframework/samples/petclinic/PetClinicApplication.java [hunk 5]
```diff
@@ -26,6 +27,7 @@
  *
  */
 @SpringBootApplication
+@EnableScheduling
```
Comment (jadmalek) [inline src/main/java/org/springframework/samples/petclinic/PetClinicApplication.java:30]: Is this annotation necessary? Same applies to the @component one below

File: src/main/java/org/springframework/samples/petclinic/owner/PetType.java [hunk 6]
```diff
@@ -74,7 +151,7 @@ public void forklift() {
         try {
             CSVReader reader = new CSVReader(new FileReader("new-datastore/pet-types.csv"));
 
-            Connection conn = DriverManager.getConnection("jdbc:mysql://localhost:3306/petclinic", "root", "root");
+            Connection conn = DriverManager.getConnection("jdbc:mysql://localhost:3306/petclinic?autoReconnect=true&useSSL=false", "root", "root");
```
Comment (jadmalek) [inline src/main/java/org/springframework/samples/petclinic/owner/PetType.java]: Is this change necessary for the introduction of the hash checker?

File: src/main/java/org/springframework/samples/petclinic/PetClinicApplication.java (hunk 5 above)
Comment (amawai) [reply src/main/java/org/springframework/samples/petclinic/PetClinicApplication.java:30]: According to the Spring documents, it's necessary for scheduled tasks to run properly, at least for the `@EnableScheduling`

File: src/main/java/org/springframework/samples/petclinic/owner/OwnerRepositoryCSV.java (hunk 2 above)
Comment (mnhn329) [reply src/main/java/org/springframework/samples/petclinic/owner/OwnerRepositoryCSV.java]: None, that was leftover from a previous iteration of the function

File: src/main/java/org/springframework/samples/petclinic/owner/ConsistencyChecker.java (hunk 4 above)
Comment (mnhn329) [reply src/main/java/org/springframework/samples/petclinic/owner/ConsistencyChecker.java]: No, not anymore

File: src/main/java/org/springframework/samples/petclinic/owner/OwnerRepositoryCSV.java (hunk 1 above)
Comment (mnhn329) [reply src/main/java/org/springframework/samples/petclinic/owner/OwnerRepositoryCSV.java]: thats logical, itll be changed

File: src/main/java/org/springframework/samples/petclinic/owner/PetType.java (hunk 6 above)
Comment (mnhn329) [reply src/main/java/org/springframework/samples/petclinic/owner/PetType.java]: No, it is was an unrelated and inconsequential change

**Review by jadmalek — APPROVED** [review]
Solid implementation and fixes after comments. As long as it works as expected, this can be integrated with the rest of the migration testing.

**Review by amawai — APPROVED** [review]
